sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -0,0 +1,1233 @@
+                    batch_loss, coef_grads, intercept_grads = self._backprop(
+                        X[batch_slice], y[batch_slice],
+                        activations, deltas, coef_grads,
+                        intercept_grads)
+                    accumulated_loss += batch_loss * (batch_slice.stop -
+                                                      batch_slice.start)
+
+                    # update weights
+                    grads = coef_grads + intercept_grads
+                    self._optimizer.update_params(grads)
+
+                self.n_iter_ += 1
+                self.loss_ = accumulated_loss / X.shape[0]
+
+                self.t_ += n_samples
+                self.loss_curve_.append(self.loss_)
+                if self.verbose:
+                    print("Iteration %d, loss = %.8f" % (self.n_iter_,
+                                                         self.loss_))
+
+                # update no_improvement_count based on training loss or
+                # validation score according to early_stopping
+                self._update_no_improvement_count(early_stopping, X_val, y_val)
+
+                # for learning rate that needs to be updated at iteration end
+                self._optimizer.iteration_ends(self.t_)
+
+                if self._no_improvement_count > 2:
+                    # not better than last two iterations by tol.
+                    # stop or decreate learning rate
+                    if early_stopping:
+                        msg = ("Validation score did not improve more than "
+                               "tol=%f for two consecutive epochs." % self.tol)
+                    else:
+                        msg = ("Training loss did not improve more than tol=%f"
+                               " for two consecutive epochs." % self.tol)
+
+                    is_stopping = self._optimizer.trigger_stopping(
+                        msg, self.verbose)
+                    if is_stopping:
+                        break
+                    else:
+                        self._no_improvement_count = 0
+
+                if incremental:
+                    break
+
+                if self.n_iter_ == self.max_iter:
+                    warnings.warn('SGD: Maximum iterations have reached and'
+                                  ' the optimization hasn\'t converged yet.'
+                                  % (), ConvergenceWarning)
+        except KeyboardInterrupt:
+            pass
+
+        if early_stopping:
+            # restore best weights
+            self.coefs_ = self._best_coefs
+            self.intercepts_ = self._best_intercepts
+
+    def _update_no_improvement_count(self, early_stopping, X_val, y_val):
+        if early_stopping:
+            # compute validation score, use that for stopping
+            self.validation_scores_.append(self.score(X_val, y_val))
+
+            if self.verbose:
+                print("Validation score: %f" %
```

Comment:
Yes it fits in one line though looks long. I was also wondering why it didn't trigger a warning in my editor, but it turned out that the length is less than 80 characters.